dev/com.ibm.ws.security.wim.adapter.ldap_fat.krb5.1/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/krb5/KerberosContainer.java

```diff
@@ -0,0 +1,165 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.security.wim.adapter.ldap.fat.krb5;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.github.dockerjava.api.model.ExposedPort;
+import com.github.dockerjava.api.model.InternetProtocol;
+import com.github.dockerjava.api.model.PortBinding;
+import com.github.dockerjava.api.model.Ports;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.containers.ExternalDockerClientFilter;
+import componenttest.containers.SimpleLogConsumer;
+import componenttest.custom.junit.runner.FATRunner;
+
+public class KerberosContainer extends GenericContainer<KerberosContainer> {
+
+    private static final Class<?> c = KerberosContainer.class;
+
+    public static final String KRB5_REALM = "EXAMPLE.COM";
+    public static final String KRB5_KDC = "kerberos";
+    public static final String KRB5_PASS = "admin";
+
+    // NOTE: If this is ever updated, don't forget to push to docker hub, but DO NOT overwrite existing versions
+    private static final String IMAGE = "zachhein/krb5-server:0.1";
+
+    private int tcp_88;
+
+    public KerberosContainer(Network network) {
+        super(IMAGE);
+        withNetwork(network);
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    protected void configure() {
+        //withExposedPorts(99, 464, 749);
+        withNetworkAliases(KRB5_KDC);
+        withCreateContainerCmdModifier(cmd -> {
+            cmd.withHostName(KRB5_KDC);
+        });
+
+        String hostname = ExternalDockerClientFilter.instance().getHostname();
```

Comment:
You might want to also check the ExternalDockerClientFilter is valid here otherwise you might get a null value for hostname:

```suggestion
        if (ExternalDockerClientFilter.instance().isValid()) {
        String hostname = ExternalDockerClientFilter.instance().getHostname();
        }
```

Reply:
good call, I think that is easier to understand than
if (!"false".equals(usingRemoteDocker))